### ISSUE TITLE
CPU leak in latest Tiled 1.2 snapshot

### ISSUE THREAD

**Issue by tomcashman** [bjorn/tiled#1990]
If I leave Tiled open with the world editing mode, after some time (maybe like 30 minutes to an hour?), Tiled ends up using 90% of my CPU. Currently using Mac OS X 10.13.6 and had to force quit the app twice this evening.

**Comment by bjorn** [comment]
Did this never happen with earlier versions? Does the problem go away if you take an older snapshot? (if you installed using the itch.io app, you should be able to switch to an older version there - if that is not a new feature in "[kitch](https://fasterthanlime.itch.io/kitch-setup)"). I'm just asking because I can't immediately think of any change in the latest snapshot that could cause such high CPU usage.

I'll try to reproduce the issue in a debugger on my Mac next week (could take till Wednesday). If it is reproducible, it would be interesting to pause the app and see what is keeping it so busy by looking at the stack trace.

**Comment by tomcashman** [comment]
I haven't had a chance to try the old snapshot - preparing for Gamescom at the moment so probably won't get a chance to test until after Gamescom.

But the issue has been consistent last few days so it's reproducible.

**Comment by bjorn** [comment]
Alright, I'll try to run into it as well.

Talking about Gamescom, are you there on Tuesday? If so we should definitely meet up!

**Comment by tomcashman** [comment]
Yup I'm there all week showing Alchemic Cutie as part of the Indie Arena Booth. I'll send you an email to arrange a time :)

**Comment by bjorn** [comment]
So, thanks to @tomcashman for bringing his laptop so we could reproduce this issue! Turns out it was due to tile animation being enabled, which takes a heavy toll on the CPU when you're zoomed out on either a large map, or in this case multiple smaller maps.

This means it's at least not a regression and there is an easy workaround (just disabling playback of tile animations). But of course it would be nice to look into optimizing the rendering update.
